internal/interface/grpc/handlers/broker.go

```diff
@@ -41,6 +45,55 @@ func (l *listener[T]) includesAny(topics []string) bool {
 	return false
 }
 
+func (l *listener[T]) addTopics(topics []string) {
+	l.lock.Lock()
+	defer l.lock.Unlock()
+	if l.topics == nil {
+		l.topics = make(map[string]struct{}, len(topics))
+	}
+	for _, topic := range topics {
+		l.topics[formatTopic(topic)] = struct{}{}
+	}
+}
+
+func (l *listener[T]) removeTopics(topics []string) {
+	l.lock.Lock()
+	defer l.lock.Unlock()
+	if l.topics == nil {
+		return
+	}
+	for _, topic := range topics {
+		delete(l.topics, formatTopic(topic))
+	}
+}
+
+func (l *listener[T]) overwriteTopics(topics []string) {
+	l.lock.Lock()
+	defer l.lock.Unlock()
+	newTopics := make(map[string]struct{}, len(topics))
+	for _, topic := range topics {
+		newTopics[formatTopic(topic)] = struct{}{}
+	}
+	l.topics = newTopics
+}
+
+func (l *listener[T]) getTopics() []string {
+	l.lock.RLock()
+	defer l.lock.RUnlock()
+	out := make([]string, 0, len(l.topics))
+	for t := range l.topics {
+		out = append(out, t)
+	}
+	return out
+}
+
+func (l *listener[T]) channel() chan T {
+	// no strong reason to lock here, but keep RLock to be safe if ch could be replaced
+	l.lock.RLock()
+	defer l.lock.RUnlock()
+	return l.ch
+}
```

Comment:
better to revert and drop this function. The lock is used to handle concurrent access to `l.topics`. If we ever need to handle concurrent access to the channel (and it seems not to be the case here) we'll add a dedicated mutex.